Entitlements are not correctly applied in Simulator

pvollan opened the pull request:
#### d94c1b592f6e84dce7a38cfd69818aa49bcb9cc3
<pre>
Entitlements are not correctly applied in Simulator
<a href="https://bugs.webkit.org/show_bug.cgi?id=268162">https://bugs.webkit.org/show_bug.cgi?id=268162</a>
<a href="https://rdar.apple.com/121541359">rdar://121541359</a>

Reviewed by Brent Fulgham and Elliott Williams.

Entitlements are not correctly applied in Simulator for WebKit extensions. This patch is
addressing this by setting CODE_SIGN_ENTITLEMENTS for Simulator, and returning early from
process-entitlements.sh, since running the script is not needed in the Simulator.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/273606@main">https://commits.webkit.org/273606@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/27a5a75052ce59ee9d6b85abbc3be97b0a45a23c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35932 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14881 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38125 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38648 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32322 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17293 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11904 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31070 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36485 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12577 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31912 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11041 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/36112 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11055 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39898 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32646 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32417 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37009 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11227 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9139 "Found 1 new test failure: http/tests/inspector/dom/didFireEvent.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35092 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12974 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11739 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4665 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12057 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
This seems reasonable, but I'm not an expert. I suggest having @emw-apple take a look, too.

emw-apple commented on Source/WebKit/Configurations/BaseExtension.xcconfig:
It makes sense that we should not be using `--entitlements` as a code signing flag, since simulated entitlements need to be baked into the binary. But the build system does that automatically when you set CODE_SIGN_ENTITLEMENTS.

In the XPC service targets, we set `CODE_SIGN_INJECT_BASE_ENTITLEMENTS = YES` and set the path in `CODE_SIGN_ENTITLEMENTS`: https://github.com/WebKit/WebKit/blob/f1cd53b0a8e7d197d0f58fd6bab9d072f8d234ff/Source/WebKit/Configurations/BaseXPCService.xcconfig#L84-L89

I don't see why the situation is any different here, can we do this instead of the manual ld flags?

pvollan replied:
Ah, good point, I will look into that!

Thanks for reviewing!

pvollan committed:
Entitlements are not correctly applied in Simulator
https://bugs.webkit.org/show_bug.cgi?id=268162
rdar://121541359

Reviewed by Brent Fulgham and Elliott Williams.

Entitlements are not correctly applied in Simulator for WebKit extensions. This patch is
addressing this by setting CODE_SIGN_ENTITLEMENTS for Simulator, and returning early from
process-entitlements.sh, since running the script is not needed in the Simulator.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: https://commits.webkit.org/273606@main
